cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -291,6 +301,58 @@ private static String getMetadata(String resource) throws IOException {
     }
   }
 
+  /**
+   * Recursively deletes all the contents of the directory and the directory itself.
+   */
+  public static void deleteDirectoryContents(@Nullable File file) {
+    if (file == null) {
+      return;
+    }
+
+    if (file.isDirectory()) {
+      File[] entries = file.listFiles();
+      if (entries != null) {
+        for (File entry : entries) {
+          deleteDirectoryContents(entry);
+        }
+      }
+    }
+    if (!file.delete()) {
+      LOG.warn("Failed to delete file {}.", file);
+    }
+  }
+
+  /**
+   * Recursively deletes all the contents of the directory and the directory itself with retries.
+   */
+  public static void deleteDirectoryWithRetries(@Nullable File file)  {
```

Comment:
Shouldn't this be synchronized since multiple concurrent invalidations can go on?

Reply:
Once synchronized, better to also return immediately if (!Files.exists(DataprocUtils.CACHE_DIR_PATH))

Reply:
ack, `deleteDirectoryContents` method does a `file.isDirectory()` check first which will immediately return false if it is does not exist.